pkg/member/member.go

```diff
@@ -185,12 +185,16 @@ func (m *EmbeddedEtcdMember) CampaignLeader(ctx context.Context, leaseTimeout in
 	failpoint.Inject("skipCampaignLeaderCheck", func() {
 		failpoint.Return(m.leadership.Campaign(leaseTimeout, m.MemberValue()))
 	})
+
 	if m.leadership.GetCampaignTimesNum() >= campaignLeaderFrequencyTimes {
-		log.Warn("campaign times is too frequent, resign and campaign again",
-			zap.String("leader-name", m.Name()), zap.String("leader-key", m.GetLeaderPath()))
 		m.leadership.ResetCampaignTimes()
-		return m.ResignEtcdLeader(ctx, m.Name(), "")
+		if err := m.ResignEtcdLeader(ctx, m.Name(), ""); err != nil {
+			return err
+		}
+		return errors.Errorf("campaign times is too frequent, resign and campaign again, "+
```

Comment:
Do we need to define it in `pkg/errs/errno.go`?

Reply:
good idea, how about `ErrEtcdLeaderFrequentlyChange`